thunder/executors/nvfuserex_impl.py

```diff
@@ -786,6 +786,22 @@ def _can_fuse_node(n: Node):
         #   (Used to name fusions like nvFusion0, nvFusion1, ...)
         fusion_counter: int = 0
         for bsyms in bound_symbol_groups:
+            # Related to in-place ops.
+            # prims.copy_ is a no-op for NVFuser in a sense that it does not
+            # generate nor runs any kernels.
+            # For that reason we should avoid fusing prims.copy_ unless
+            # it comes after other non-copy symbols in a fusion.
+            # See the following relevant issues:
+            # https://github.com/Lightning-AI/lightning-thunder/issues/789
+            # https://github.com/Lightning-AI/lightning-thunder/issues/791
+            # NOTE: filter all first "dangling" no-op copies
+            while len(bsyms) > 0 and bsyms[0].sym.id is prims.PrimIDs.COPY_:
+                fused_bsyms.append(bsyms[0])
+                bsyms = bsyms[1:]
```

Comment:
* It seems we're looking at the absolute order of operations, rather than a dataflow order. But maybe we don't have infra to do so via dataflow?
* Should logic like this go into `_should_fuse`, above, instead of here? @jjsjann123

Reply:
Oh, side note: could I ask you to have the comment define what "dangling" means here?

Reply:
I had a similar idea. Tried the easiest solution first

Reply:
Yeah, that's exactly the kind of thing I was worried about :-)

As you're highlighting, we have a lot of implicit assumptions here on the sorting of ops because we are not relying on dataflow. Such a requirement is implicit on earlier code's behavior, and there's really not a strong reason for that earlier behavior to exist (at least before this).

This is a good definition of fragility--when disconnected, distant pieces of code impose specific requirements on one another.

One of the reasons to ping Jie to get his thoughts is that maybe this fragility is fine here: if our thought is that we're going to scrap much of the fusion logic entirely in a rewrite, a little bit of extra debt here probably is not worth worrying about.